gapis/api/vulkan/frame_loop.go

```diff
@@ -1326,6 +1330,61 @@ func (f *frameLoop) resetPipelineLayouts(ctx context.Context, stateBuilder *stat
 		stateBuilder.createPipelineLayout(GetState(f.loopStartState).pipelineLayouts.Get(toCreate))
 	}
 
+	// The shadow state for PipelineLayouts does not contain reference to the ComputePipelines they are used in. So we have to loop around finding the story.
+	for _, computePipelineObject := range GetState(f.loopStartState).ComputePipelines().All() {
+		pipelineLayout := computePipelineObject.PipelineLayout()
+		if _, ok := f.pipelineLayoutToCreate[pipelineLayout.VulkanHandle()]; ok {
+			f.pipelineLayoutToDestroy[pipelineLayout.VulkanHandle()] = true
+			f.pipelineLayoutToCreate[pipelineLayout.VulkanHandle()] = true
+		}
+	}
+
+	// The shadow state for PipelineLayouts does not contain reference to the GraphicsPipelines they are used in. So we have to loop around finding the story
+	for _, graphicsPipelineObject := range GetState(f.loopStartState).GraphicsPipelines().All() {
+		pipelineLayout := graphicsPipelineObject.Layout()
```

Comment:
On a compute pipeline you call `.PipelineLayout()`, whereas on a graphics pipeline you call `.Layout()`, I would have expected to see `.PipelineLayout()` in both cases. Can you confirm this is doing the correct thing here?

Reply:
Yup. The names in the API files are not always consistent sadly.